bazel_tools/client_server_test/runner/Main.hs

```diff
@@ -42,33 +37,31 @@ readPortFile n file =
       readPortFile (n-1) file
     Just p -> pure p
 
-runCheckedProc :: T.Text -> CreateProcess -> IO () -> IO ()
```

Comment:
What was the purpose of this code before? Deleting it seems reasonable, but do you understand why someone did it this way in the first place? Was it from the days we sniffed the port on stdout?

Reply:
It seems to be in such form from the day 1. The main difference from `withCreateProcess` I can see is that server logs are prefixed with `SERVER: ` making the test output a bit easier to read as it can be distinguished from client process' stdout/stderr.

Reply:
It seems hard to justify that level of additional complexity for a test.

Reply:
Actually this part was problematic on Windows as conduits copying stdout/stderr where not getting killed, blocking the runner https://github.com/digital-asset/daml/blob/536b1ea87fe763c278a6dc1caf435de09db77c8c/bazel_tools/client_server_test/runner/Main.hs#L48

Reply:
In the integration tests, we address this by calling `terminateProcess` at the end https://github.com/digital-asset/daml/blob/c66211d0345250f5bbac8fdf2f171a27f96cd4c3/daml-assistant/integration-tests/src/Main.hs#L287

Reply:
Thanks @cocreature . It doesn't seem to be needed in this case though after removal of `withAsync` bit.